ss/pebbledb/db.go

```diff
@@ -184,13 +233,90 @@ func (db *Database) ApplyChangeset(version int64, cs *proto.NamedChangeSet) erro
 	return b.Write()
 }
 
-// Prune for the PebbleDB SS backend is currently not supported. It seems the only
-// reliable way to prune is to iterate over the desired domain and either manually
-// tombstone or delete. Either way, the operation would be timely.
-//
-// See: https://github.com/cockroachdb/cockroach/blob/33623e3ee420174a4fd3226d1284b03f0e3caaac/pkg/storage/mvcc.go#L3182
+// Prune attempts to prune all versions up to and including the current version
+// Get the range of keys, manually iterate over them and delete them
 func (db *Database) Prune(version int64) error {
-	panic("not implemented!")
+	earliestVersion := version + 1 // we increment by 1 to include the provided version
+
+	itr, err := db.storage.NewIter(nil)
+	if err != nil {
+		return err
+	}
+	defer itr.Close()
+
+	batch := db.storage.NewBatch()
+	defer batch.Close()
+
+	var (
+		counter                                 int
+		prevKey, prevKeyEncoded, prevValEncoded []byte
+		prevVersionDecoded                      int64
+	)
+
+	for itr.First(); itr.Valid(); {
+		currKeyEncoded := slices.Clone(itr.Key())
```

Comment:
I don't think we need to copy. This could improve the performance of an already very slow method.

Reply:
We actually do need this, we also need a slices.Clone(itr.Value()) below too. We ran into an issue on a node (not locally though) where this wasn't done and the iterator key changed